### ISSUE TITLE
(ec2): adding egress rules to looked up alb removes other existing egress rules

### ISSUE THREAD

**Issue by lorefnon** [aws/aws-cdk#16580]
I am attempting to introduce CF+CDK gradually into custom managed infrastructure. 

My abbreviated code looks like this: 

```ts
const albSecurityGroup = ec2.SecurityGroup.fromLookup(
    stack,
    "ALBSecurityGroup",
    "securityGroupId"
);

const securityGroup = new ec2.SecurityGroup(stack, "SecurityGroup", {
    description: "Security Group for ECS Instance",
    vpc,
    allowAllOutbound: true,
});

const vpc = ec2.Vpc.fromLookup(stack, "VPC", { vpcId: "somevpcid" });

[8080, 9090].forEach((port) => {
    albSecurityGroup.addEgressRule(securityGroup, ec2.Port.tcp(port));
    securityGroup.addIngressRule(albSecurityGroup, ec2.Port.tcp(port));
});
```

My intent is to lookup a shared ALB and add egress rules to it for a service defined in my cdk stack.

### What did you expect to happen?

Egress rules to be added to albSecurityGroup and other previous egress rules (defined manually) to remain as is.

### What actually happened?

Existing egress rules in albSecurityGroup are replaced by the ones defined in stack.

Destroying the stack restores the original egress rules though. 

### Environment

  - **CDK CLI Version  :** 1.123.0
  - **Framework Version:** 1.123.0
  - **Node.js Version:** 16.6.2
  - **OS               :** Ubuntu
  - **Language (Version):** TS 3.9.7 

--- 

I am not 100 % sure that this is a bug, but the behavior seems counter-intuitive.

**Comment by njlynch** [comment]
I believe this is a duplicate of #15297.

tl;dr - I believe the short answer is that this case occurs when the existing security group has no explicit rules defined. With no rules defined, the security group has a default behavior. However, as soon as new ingress/egress rules are defined -- in this case via a new stack -- those new rules take precedence, overriding the default behavior.

This is a quirk of how Security Groups + CloudFormation interact, so options to fix from the CDK side are limited. That being said, this has already been reported in #15297, so closing this to track the solution there.

**Comment by lorefnon** [comment]
@njlynch  We do have explicitly defined rules that allow all outbound traffic. 

When imported in a CDK template, these explicitly defined rules egress are removed in favor of the ones specified in the stack. They are restored back when the cloudformation stack is destroyed.
